prov/gni/test/api_cq.c

```diff
@@ -291,6 +293,47 @@ void api_cq_recv_allowed(ssize_t sz, uint64_t flags, char *fn)
 TestSuite(api_cq, .init = api_cq_setup, .fini = api_cq_teardown,
 	  .disabled = false);
 
+void api_cq_wait1(struct fid_cq *cq0, uint64_t cq_bind_flags)
+{
+	int ret;
+	struct fi_cq_tagged_entry cqe;
+
+	if (!cq_bind_flags)
+		return;
+
+	while ((ret = fi_cq_read(msg_cq[0], &cqe, 1)) == -FI_EAGAIN) {
+		pthread_yield();
+	}
+}
+
+void api_cq_wait2(struct fid_cq *cq0, struct fid_cq *cq1,
+		  uint64_t check_send, uint64_t check_rcv)
+{
+	int ret;
+	int source_done = 0, dest_done = 0;
+	struct fi_cq_tagged_entry s_cqe, d_cqe;
+
+	if (!check_send)
+		source_done = 1;
+
+	if (!check_rcv)
+		dest_done = 1;
+
+	do {
+		ret = fi_cq_read(cq0, &s_cqe, 1);
+		if (ret == 1) {
+			source_done = 1;
+		}
+
+		ret = fi_cq_read(cq1, &d_cqe, 1);
+		if (ret == 1) {
+			dest_done = 1;
+		}
+
+		pthread_yield();
```

Comment:
Do you need to use pthread_yield here? Since this is not a kernel module where busy spinning is frowned upon it shouldn't impact the application in user space.